File: dataset/codenet/Python/Code/s923405856.py
while True:
	n = int(input())
	
	if n == 0:
		break
	
	res = -1111111111
	s = 0
	for i in range(n):
		a = int(input())
		s = max(s + a, a)
		res = max(s, res)
	
	print(res)

File: output/llama_4_maverick/codenet/algo_based_translation/python/java/successful_translation/s923405856.java
import java.util.Scanner;

public class s923405856 {
    public static void max_subarray_sum() {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            int n = scanner.nextInt();
            if (n == 0) {
                break;
            }
            int res = Integer.MIN_VALUE;
            int s = 0;
            for (int i = 0; i < n; i++) {
                int a = scanner.nextInt();
                s = Math.max(s + a, a);
                res = Math.max(s, res);
            }
            System.out.println(res);
        }
        scanner.close();
    }

    public static void main(String[] args) {
        max_subarray_sum();
    }
}
